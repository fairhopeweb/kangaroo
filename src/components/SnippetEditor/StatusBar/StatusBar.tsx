import React, { memo } from 'react';
import classNames from 'classnames';
import { Tooltip, Button, HTMLSelect, Position, Icon, Navbar } from '@blueprintjs/core';

import TagBar from 'components/SnippetEditor/StatusBar/TagBar/TagBar';

import { languages } from 'models/languages';

import './StatusBar.scss';

type StatusBarProps = {
  snippetLanguage?: string;
  snippetTags?: string;
  snippetSource?: string;
  onShowGutter: () => void;
  onTagChange: (tag: string, remove: boolean) => void;
  onLanguageChange: (event: React.ChangeEvent<HTMLSelectElement>) => void;
};

const languageItems = Object.entries(languages).map(([key, value]) => ({ label: value.label, value: key }));

const StatusBar: React.FC<StatusBarProps> = ({
  snippetLanguage,
  snippetTags,
  snippetSource,
  onShowGutter,
  onTagChange,
  onLanguageChange,
}) => (
  <div className="StatusBar--container">
    <Tooltip content="Show gutter" position={Position.TOP}>
      <Button
        small={true}
        icon="list-detail-view"
        minimal={true}
        className="StatusBar--show-gutter"
        onClick={onShowGutter}
      />
    </Tooltip>

    {snippetSource && (
      <>
        <TagBar tags={snippetTags} onSelect={onTagChange} className="StatusBar--tag-bar" />
        <Navbar.Divider />

        <div className="StatusBar--synchronization-icon-container">
          <Icon icon="cloud" className="StatusBar--synchronization-icon" />
          <div className={classNames('StatusBar--synchronization-icon-status', snippetSource)} />
        </div>
        <Navbar.Divider />
      </>
    )}

    {snippetLanguage ? (
      <HTMLSelect
        value={snippetLanguage}
        minimal={true}
        iconProps={{ icon: 'caret-down' }}
        className="StatusBar--language-selector"
        onChange={onLanguageChange}
        options={languageItems}
      />
    ) : (
      <HTMLSelect minimal={true} iconProps={{ icon: 'caret-down' }} disabled={true} />
    )}
  </div>
);

export default memo(StatusBar);
