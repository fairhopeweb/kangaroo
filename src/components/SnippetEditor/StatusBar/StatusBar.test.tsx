import { configure, shallow } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import renderer from 'react-test-renderer';

import StatusBar from './StatusBar';

configure({ adapter: new Adapter() });

describe('<StatusBar />', () => {
  const mockProps = {
    snippetLanguage: 'text',
    snippetTags: '',
    snippetSource: 'local',
    onShowGutter: jest.fn(),
    onTagChange: jest.fn(),
    onLanguageChange: jest.fn(),
  };

  it('render without crashing', () => {
    shallow(<StatusBar {...mockProps} />);
  });

  it('matches snapshot', () => {
    const wrapper = renderer.create(<StatusBar {...mockProps} />);
    expect(wrapper).toMatchSnapshot();
  });
});
