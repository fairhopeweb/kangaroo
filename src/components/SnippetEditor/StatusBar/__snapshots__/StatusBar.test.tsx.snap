// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StatusBar /> matches snapshot 1`] = `
<div
  className="container"
>
  <span
    aria-haspopup="true"
    className="bp3-popover2-target"
    onBlur={[Function]}
    onContextMenu={[Function]}
    onFocus={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <button
      className="bp3-button bp3-minimal bp3-small showGutter"
      onBlur={[Function]}
      onClick={[MockFunction]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        aria-hidden={true}
        className="bp3-icon bp3-icon-list-detail-view"
        icon="list-detail-view"
      >
        <svg
          data-icon="list-detail-view"
          height={16}
          viewBox="0 0 16 16"
          width={16}
        >
          <path
            d="M6 9H1c-.55 0-1 .45-1 1s.45 1 1 1h5c.55 0 1-.45 1-1s-.45-1-1-1zm0 4H1c-.55 0-1 .45-1 1s.45 1 1 1h5c.55 0 1-.45 1-1s-.45-1-1-1zm9-12h-5c-.55 0-1 .45-1 1v12c0 .55.45 1 1 1h5c.55 0 1-.45 1-1V2c0-.55-.45-1-1-1zM6 5H1c-.55 0-1 .45-1 1s.45 1 1 1h5c.55 0 1-.45 1-1s-.45-1-1-1zm0-4H1c-.55 0-1 .45-1 1s.45 1 1 1h5c.55 0 1-.45 1-1s-.45-1-1-1z"
            fillRule="evenodd"
          />
        </svg>
      </span>
    </button>
  </span>
  <div
    className="container"
  >
    <div
      className="wrapper"
    >
      <div
        className="tag"
        onClick={[Function]}
        style={
          Object {
            "background": "#F55656",
          }
        }
      >
        <span
          aria-hidden={true}
          className="bp3-icon bp3-icon-small-plus actionIcon"
          icon="small-plus"
        >
          <svg
            data-icon="small-plus"
            height={16}
            viewBox="0 0 16 16"
            width={16}
          >
            <path
              d="M11 7H9V5c0-.55-.45-1-1-1s-1 .45-1 1v2H5c-.55 0-1 .45-1 1s.45 1 1 1h2v2c0 .55.45 1 1 1s1-.45 1-1V9h2c.55 0 1-.45 1-1s-.45-1-1-1z"
              fillRule="evenodd"
            />
          </svg>
        </span>
      </div>
    </div>
    <div
      className="wrapper"
    >
      <div
        className="tag"
        onClick={[Function]}
        style={
          Object {
            "background": "#F29D49",
          }
        }
      >
        <span
          aria-hidden={true}
          className="bp3-icon bp3-icon-small-plus actionIcon"
          icon="small-plus"
        >
          <svg
            data-icon="small-plus"
            height={16}
            viewBox="0 0 16 16"
            width={16}
          >
            <path
              d="M11 7H9V5c0-.55-.45-1-1-1s-1 .45-1 1v2H5c-.55 0-1 .45-1 1s.45 1 1 1h2v2c0 .55.45 1 1 1s1-.45 1-1V9h2c.55 0 1-.45 1-1s-.45-1-1-1z"
              fillRule="evenodd"
            />
          </svg>
        </span>
      </div>
    </div>
    <div
      className="wrapper"
    >
      <div
        className="tag"
        onClick={[Function]}
        style={
          Object {
            "background": "#F2B824",
          }
        }
      >
        <span
          aria-hidden={true}
          className="bp3-icon bp3-icon-small-plus actionIcon"
          icon="small-plus"
        >
          <svg
            data-icon="small-plus"
            height={16}
            viewBox="0 0 16 16"
            width={16}
          >
            <path
              d="M11 7H9V5c0-.55-.45-1-1-1s-1 .45-1 1v2H5c-.55 0-1 .45-1 1s.45 1 1 1h2v2c0 .55.45 1 1 1s1-.45 1-1V9h2c.55 0 1-.45 1-1s-.45-1-1-1z"
              fillRule="evenodd"
            />
          </svg>
        </span>
      </div>
    </div>
    <div
      className="wrapper"
    >
      <div
        className="tag"
        onClick={[Function]}
        style={
          Object {
            "background": "#15B371",
          }
        }
      >
        <span
          aria-hidden={true}
          className="bp3-icon bp3-icon-small-plus actionIcon"
          icon="small-plus"
        >
          <svg
            data-icon="small-plus"
            height={16}
            viewBox="0 0 16 16"
            width={16}
          >
            <path
              d="M11 7H9V5c0-.55-.45-1-1-1s-1 .45-1 1v2H5c-.55 0-1 .45-1 1s.45 1 1 1h2v2c0 .55.45 1 1 1s1-.45 1-1V9h2c.55 0 1-.45 1-1s-.45-1-1-1z"
              fillRule="evenodd"
            />
          </svg>
        </span>
      </div>
    </div>
    <div
      className="wrapper"
    >
      <div
        className="tag"
        onClick={[Function]}
        style={
          Object {
            "background": "#2B95D6",
          }
        }
      >
        <span
          aria-hidden={true}
          className="bp3-icon bp3-icon-small-plus actionIcon"
          icon="small-plus"
        >
          <svg
            data-icon="small-plus"
            height={16}
            viewBox="0 0 16 16"
            width={16}
          >
            <path
              d="M11 7H9V5c0-.55-.45-1-1-1s-1 .45-1 1v2H5c-.55 0-1 .45-1 1s.45 1 1 1h2v2c0 .55.45 1 1 1s1-.45 1-1V9h2c.55 0 1-.45 1-1s-.45-1-1-1z"
              fillRule="evenodd"
            />
          </svg>
        </span>
      </div>
    </div>
  </div>
  <div
    className="bp3-navbar-divider"
  />
  <span
    aria-haspopup="true"
    className="bp3-popover2-target"
    onBlur={[Function]}
    onContextMenu={[Function]}
    onFocus={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="synchronizationIconContainer"
      tabIndex={0}
    >
      <span
        aria-hidden={true}
        className="bp3-icon bp3-icon-cloud synchronizationIcon"
        icon="cloud"
      >
        <svg
          data-icon="cloud"
          height={16}
          viewBox="0 0 16 16"
          width={16}
        >
          <path
            d="M12 6c-.03 0-.07 0-.1.01A5 5 0 002 7c0 .11.01.22.02.33A3.51 3.51 0 000 10.5C0 12.43 1.57 14 3.5 14H12c2.21 0 4-1.79 4-4s-1.79-4-4-4z"
            fillRule="evenodd"
          />
        </svg>
      </span>
      <div
        className="synchronizationIconStatus local"
      />
    </div>
  </span>
  <div
    className="bp3-navbar-divider"
  />
  <div
    className="bp3-html-select bp3-minimal languageSelector"
  >
    <select
      multiple={false}
      onChange={[MockFunction]}
      value="text"
    >
      <option
        label="Text"
        value="text"
      >
        Text
      </option>
      <option
        label="JavaScript"
        value="javascript"
      >
        JavaScript
      </option>
      <option
        label="TypeScript"
        value="typescript"
      >
        TypeScript
      </option>
      <option
        label="Java"
        value="java"
      >
        Java
      </option>
      <option
        label="XML"
        value="xml"
      >
        XML
      </option>
      <option
        label="CSS"
        value="css"
      >
        CSS
      </option>
      <option
        label="SCSS"
        value="scss"
      >
        SCSS
      </option>
      <option
        label="Markdown"
        value="markdown"
      >
        Markdown
      </option>
      <option
        label="SQL"
        value="sql"
      >
        SQL
      </option>
      <option
        label="JSON"
        value="json"
      >
        JSON
      </option>
      <option
        label="HTML"
        value="html"
      >
        HTML
      </option>
      <option
        label="C#"
        value="csharp"
      >
        C#
      </option>
    </select>
    <span
      className="bp3-icon bp3-icon-caret-down"
      icon="caret-down"
    >
      <svg
        data-icon="caret-down"
        height={16}
        viewBox="0 0 16 16"
        width={16}
      >
        <desc>
          Open dropdown
        </desc>
        <path
          d="M12 6.5c0-.28-.22-.5-.5-.5h-7a.495.495 0 00-.37.83l3.5 4c.09.1.22.17.37.17s.28-.07.37-.17l3.5-4c.08-.09.13-.2.13-.33z"
          fillRule="evenodd"
        />
      </svg>
    </span>
  </div>
</div>
`;
